src/nbl/asset/interchange/IAssetLoader.cpp

```diff
@@ -35,6 +35,24 @@ void IAssetLoader::IAssetLoaderOverride::insertAssetIntoCache(SAssetBundle& asse
         m_manager->insertAssetIntoCache(asset, ASSET_MUTABILITY_ON_CACHE_INSERT);
 }
 
+bool IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
```

Comment:
dont trust the bool, because you'll need to check yourself, so dont make a function that returns a `bool`

also return the `_chosenAsset`